pattern-library/sass/patterns/_grid.scss

```diff
@@ -14,276 +14,176 @@
 // #GALLERY - CANNED + RESPONSIVE
 // #DEBUG
 
-
 // ----------------------------
-// #SETTINGS
+// #UTILITY
 // ----------------------------
-// grid - configuration (susy-based)
-
-// set RTL/LTR specific variables
-$grid-direction-default: ltr;
-$grid-direction-reversed: rtl;
-
-@if $layout-direction == rtl {
-    $grid-direction-default: rtl;
-    $grid-direction-reversed: ltr;
+// Utility functions/mixins designed for Susy API-compatibility.
+@function cols($cols) {
+    @return $cols / $grid-column-count * 100%;
 }
 
-// susy-based configuration (see http://susydocs.oddbird.net/en/latest/settings/)
-$susy: (
-    use-custom: (
-        background-options: false,
-        box-sizing: false,
-        breakpoint: true,
-        clearfix: true,
-        rem: true
-    ),
-    debug: (
-        color: $grid-debug-color,
-    ),
-    math: fluid,
-    output: float
-);
-
-// LAYOUT: set default susy settings
-// (gutters are handled outside of elements - by default)
-$susy-default: (
-    flow: $grid-direction-default,
-    column-width: $grid-column-width,
-    columns: $grid-column-count,
-    container-position: $grid-container-position,
-    container: $grid-max-width,
-    global-box-sizing: border-box,
-    gutters: $grid-gutter-width,
-    gutter-position: $grid-gutter-position
-);
-
-// LAYOUT: set susy settings especially for manually set grids in the DOM
-// (gutters are handled inside of elements - by default)
-$susy-manual: layout($susy-default $grid-manual-gutter-position);
-
-// LAYOUT: set debug settings
-$susy-debug: layout($susy-default show);
-
-
-// ----------------------------
-// #GLOBALS
-// ----------------------------
-// set basic grid settings for app
-@include layout($susy-default);
+@mixin span($cols, $margin:'') {
+    @include margin-left(calc(0.5 * #{$grid-gutter-width}));
+    @include margin-right(calc(0.5 * #{$grid-gutter-width}));
+    width: calc(#{cols($cols)} - #{$grid-gutter-width});
+
+    @if ($margin == 'before') {
+        @include margin-right(0);
+        width: calc(#{cols($cols)} - 0.5 * #{$grid-gutter-width});
+    } @else if ($margin == 'after') {
+        @include margin-left(0);
+        width: calc(#{cols($cols)} - 0.5 * #{$grid-gutter-width});
+    } @else if ($margin == 'none') {
+        @include margin-left(0);
+        @include margin-right(0);
+        width: cols($cols);
+    }
+}
 
+@mixin pre($cols) {
+    @include margin-left(cols($cols) !important);
```

Comment:
That's a very valid anxiety, I throw up red flags when I see `!important`s too.

The `pre` and `post` mixins are used in grid columns to give them a certain amount of empty columns before or after that column. The best example of how they're used is these guys:

![](https://i.bjacobel.com/20161101-qfrmi.png)

Each of those boxes looks in Sass like this:

```sass
.col-H {
    @include span(4, none);
    @include pre(8);
}
```

which when it's compiled out to CSS looks something like this: 

```css
.col-H {
    margin-left: 0;
    margin-right: 0;
    width: 33.333%;
    margin-left: 66.666% !important;
}
```

So, as you can see, in that case, the `!important` isn't necessary - the `pre`'s `margin-left: 66%` would override the one set on the same selector at the same specificity just by virtue of coming after it in the stylesheet. But if you messed up and wrote:

```sass
    @include pre(8);
    @include span(4, none);
```

Then the `margin-left` from your `pre` would be overridden by the `margin-left: 0` from the `span` mixin, and `pre` wouldn't work at all. So I added the `!important` because 1) I wanted these mixins to be able to be used on the same selector (which maintains compatibility with the existing Susy API) and 2) so that I wouldn't create a situation where the order in which you used the mixins could have an effect.

I think the only way around this would be to use a pseudoelement `:before` with `pre()` (and `:after` with `post()`), rather than margins, but I played around with that for a little bit and couldn't figure out how to make pseudoelements work well with flexbox.